### PR TITLE
ci: Added workflow for Storybook deployment

### PR DESCRIPTION
**Pull request by albinAppsmith**
## Description

Added workflow to build static file for storybook deployment

Fixes [#14936](https://github.com/appsmithorg/appsmith/issues/14936)

## Type of change

- New feature (non-breaking change which adds functionality)

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: .github/workflows/ds-storybook-deploy.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+jobs:
+  deploy-storybook:
+    runs-on: ubuntu-latest
+    steps:
+      # Checkout the code
+      - uses: actions/checkout@v3
+
+      # Setup node with version 16.x and NPM registry url
+      - uses: actions/setup-node@v1
```
Comment (sharat87) [inline .github/workflows/ds-storybook-deploy.yml:23]: Any reason we aren't using `@v3` of this? 🤔

Comment (albinAppsmith) [reply .github/workflows/ds-storybook-deploy.yml:23]: That's the latest version

File: .github/workflows/ds-storybook-deploy.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+          registry-url: "https://registry.npmjs.org"
+
+      # yarn build to build the project
+      - name: Build package
+        run: yarn build-storybook
+
+      - name: configure aws credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: arn:aws:iam::391311798816:role/design-systems-github-actions-deploy-s3
```
Comment (rohitagarwal88) [inline .github/workflows/ds-storybook-deploy.yml:35]: arn:aws:iam::391311798816? can it be a security concern?

File: .github/workflows/ds-storybook-deploy.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+          role-session-name: ds-storybook-deploy-workflow
+          aws-region: us-east-2
```
Comment (rohitagarwal88) [inline .github/workflows/ds-storybook-deploy.yml:37]: Shall we move region git secret key?

**Review by rohitagarwal88 — CHANGES_REQUESTED** [review]

File: .github/workflows/ds-storybook-deploy.yml (hunk 2 above)
Comment (sharat87) [reply .github/workflows/ds-storybook-deploy.yml:35]: Account IDs aren't sensitive information, and should be okay here.

@mohanarpit, if your opinion on this is different, we can move it to a secret, by I don't think account ID should be a secret.

File: .github/workflows/ds-storybook-deploy.yml (hunk 3 above)
Comment (sharat87) [reply .github/workflows/ds-storybook-deploy.yml:37]: We used to do that for some other older workflows, but, creates more problems than it solves. Also, not technically a secret, so I think we should be okay with it here.

File: .github/workflows/ds-storybook-deploy.yml (hunk 1 above)
Comment (sharat87) [reply .github/workflows/ds-storybook-deploy.yml:23]: Not sure I understand. We want Node 16, so if we change this to `v3`, and change `node-version: "16.x"` to `node-version: 16`, it should work just as well? But with the current version of the setup-node action?

Comment (albinAppsmith) [reply .github/workflows/ds-storybook-deploy.yml:23]: Oh got it sorry, I got your question wrong. I will update this

**Review by rohitagarwal88 — APPROVED** [review]

**Pull request merged**